pkg/kubelet/config/apiserver.go

```diff
@@ -23,13 +23,27 @@ import (
 	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
 	"k8s.io/kubernetes/pkg/fields"
 	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
+	"k8s.io/kubernetes/pkg/runtime"
 	"k8s.io/kubernetes/pkg/types"
+	"k8s.io/kubernetes/pkg/watch"
 )
 
 // NewSourceApiserver creates a config source that watches and pulls from the apiserver.
 func NewSourceApiserver(c *clientset.Clientset, nodeName types.NodeName, updates chan<- interface{}) {
-	lw := cache.NewListWatchFromClient(c.Core().RESTClient(), "pods", api.NamespaceAll, fields.OneTermEqualSelector(api.PodHostField, string(nodeName)))
```

Comment:
It is straightforward. I can update this PR to generate the ``ListWatch(fieldSelector field.Selector)`` as well and update all the places where the ``ListWatch`` struct is populated manually. Just wondering where to put the versioned ``ListWatch`` struct. What about ``pkg/watch/v1``? There is already ``pkg/watch/versioned``, though ``v1`` looks better.